kimchi/src/circuits/constraints.rs

```diff
@@ -503,14 +561,35 @@ impl<F: PrimeField + SquareRootField> Builder<F> {
 
         // Range check constraint selector polynomials
         let range_check_selector_polys = {
-            if !circuit_gates_used.is_disjoint(&range_check::circuit_gates().into_iter().collect())
+            if !circuit_gates_used
+                .is_disjoint(&range_check::gadget::circuit_gates().into_iter().collect())
             {
-                range_check::selector_polynomials(&gates, &domain)
+                range_check::gadget::selector_polynomials(&gates, &domain)
             } else {
                 vec![]
             }
         };
 
+        // Foreign field addition constraint selector polynomial
+        let foreign_field_add_selector_poly = {
+            if !circuit_gates_used.is_disjoint(
+                &foreign_field_add::gadget::circuit_gates()
+                    .into_iter()
+                    .collect(),
+            ) {
+                Some(
+                    selector_polynomials(
```

Comment:
why use `selector_polynomials` and not `selector_polynomial`?

Reply:
If we are aiming for a more standardized format between the gates, perhaps `selector_polynomials` is more general in case some gadgets only use one gate and others use more. But yeah, I agree that in this particular case for FFAdd this is just a one element vector.

Reply:
But yeah, I agree that if we stick to selector_polynomial in constraints and not per-gate, we really do not care about the length of the array at this point to have a different function for the "plural". We can just inline the array creation.